bento_wes/runs.py

```diff
@@ -288,6 +291,53 @@ def _create_run(db, c):
         return flask_bad_request_error("Assertion error: bad run request format")
 
 
+def fetch_run_details(c, public_endpoint=False):
+    c.execute(
+        "SELECT r.id AS run_id, r.state AS state, rr.*, rl.* "
+        "FROM runs AS r, run_requests AS rr, run_logs AS rl "
+        "WHERE r.request = rr.id AND r.run_log = rl.id"
+    )
+
+    runs = []
+    for r in c.fetchall():
+        request = run_request_dict(r) if not public_endpoint else run_request_dict_public(r)
+        run_log = run_log_dict(r["run_id"], r) if not public_endpoint else run_log_dict_public(r["run_id"], r)
+        end_time = run_log['end_time']
+        state = r['state']
+
+        run_data = {
+            'run_id': r["run_id"],
+            'state': state,
+            'details': {
+                'end_time': end_time if public_endpoint else None,
+                'run_id': r["run_id"],
+                'state': state,
+                'request': request,
+                'run_log': run_log,
+                'task_logs': None if public_endpoint else get_task_logs(c, r["run_id"]),
+            }
+        }
+
+        if public_endpoint and state == 'COMPLETE':
+            runs.append(run_data)
+        elif not public_endpoint:
+            runs.append(run_data)
```

Comment:
this will not append runs which are COMPLETE if public is false, which is not desired behaviour.